gtsam/slam/tests/testEssentialMatrixFactor.cpp

```diff
@@ -351,7 +353,112 @@ TEST (EssentialMatrixFactor3, minimization) {
   EXPECT_DOUBLES_EQUAL(0, graph.error(result), 1e-4);
 }
 
-} // namespace example1
+//*************************************************************************
+TEST(EssentialMatrixFactor4, factor) {
+  Key keyE(1);
+  Key keyK(1);
+  for (size_t i = 0; i < 5; i++) {
+    EssentialMatrixFactor4<Cal3_S2> factor(keyE, keyK, pA(i), pB(i), model1);
+
+    // Check evaluation
+    Vector1 expected;
+    expected << 0;
+    Matrix HEactual;
+    Matrix HKactual;
+    Vector actual = factor.evaluateError(trueE, trueK, HEactual, HKactual);
+    EXPECT(assert_equal(expected, actual, 1e-7));
+
+    // Use numerical derivatives to calculate the expected Jacobian
+    Matrix HEexpected;
+    Matrix HKexpected;
+    typedef Eigen::Matrix<double, 1, 1> Vector1;
+    boost::function<Vector(const EssentialMatrix &, const Cal3_S2 &)> f =
+        boost::bind(&EssentialMatrixFactor4<Cal3_S2>::evaluateError, factor, _1,
+                    _2, boost::none, boost::none);
+    HEexpected = numericalDerivative21<Vector1, EssentialMatrix, Cal3_S2>(
+        f, trueE, trueK);
+    HKexpected = numericalDerivative22<Vector1, EssentialMatrix, Cal3_S2>(
+        f, trueE, trueK);
+
+    // Verify the Jacobian is correct
+    EXPECT(assert_equal(HEexpected, HEactual, 1e-8));
+    EXPECT(assert_equal(HKexpected, HKactual, 1e-8));
```

Comment:
Is this different from using EXPECT_CORRECT_FACTOR_JACOBIANS?

Reply:
Yes, if you're creating a factor, please use the macro. Change for all uses in this file.